### ISSUE TITLE
graphql and REST APIs return different results with where clauses

### ISSUE THREAD

**Issue by theuargb** [payloadcms/payload#576]
# Bug Report

Trying to get `posts` within a specific category on the demo site. `graphql` and `REST` APIs return different results (`where post in categories` doesn't work well on `graphql` side)

## Expected Behavior

`totalDocs` is the same via any API endpoint

## Current Behavior

`GET https://demo.payloadcms.com/api/posts/?where[or][0][and][0][category][in][0]=628a8832a975e5ef952b1a21` (with query-string generated by the admin panel) returns:
```
{
    docs: [...],
    hasNextPage: false,
    hasPrevPage: false,
    limit: 10,
    nextPage: null,
    page: 1,
    pagingCounter: 1,
    prevPage: null,
    totalDocs: 1, // uno!
    totalPages: 1
}
```

while
```
# Write your query or mutation here
query {
  Posts(
    where: {
      category: {
        in: ["628a8832a975e5ef952b1a21"]
      }
    }
    draft: false
    locale: en
    fallbackLocale: en
    page: 1
    limit: 10
  ) {
    docs {
      id
      title
      category {
        id
        name
      }
    }
    totalDocs
  }
}
```
returns

```
{
  "data": {
    "Posts": {
      "docs": [...],
      "totalDocs": 4 // like no filters applied
    }
  }
}
```

## Possible Solution
--

## Steps to Reproduce
1. Complete requests above

## Detailed Description

![image](https://user-images.githubusercontent.com/83121880/169713273-7fad590d-bad3-417b-81c0-c758fee50731.png)
![image](https://user-images.githubusercontent.com/83121880/169713278-ffe48e01-99b4-437d-9b58-80f68a1e7a6a.png)

**Comment by jmikrut** [comment]
Hey @theuargb — thank you for the report! There was a bug with GraphQL `where` clauses against `hasMany` `relationship` fields. Wow, what a sentence.

Published version `0.17.2` with a fix!

**Comment by dan-wearearch** [comment]
Hello, I am having the opposite result. I have the same query on both REST and Graphql, the REST query brings back one result (which is correct) and the Graphql brings back no results. 

Below are the two queries. 
## Rest
```
http://localhost:3000/api/products?where[categories][in]=64a52ef6f53cf502e2aa0cd8
```
Result
```json
{
  "docs": [
    {
      "id": "64a52fa1f53cf502e2aa0d46",
      "title": "Carvan",
      "shortDescription": "caahds ofhgaosdufgasdu gfasudgfoasydg foays dofa sodf haosdh foasdh foashd foas",
      "price": "232",
      "inYard": true,
      "categories": [
        {
          "id": "64a52ef6f53cf502e2aa0cd8",
          "name": "Car",
          "archived": false,
          "meta": {
            "meta_title": "car",
            "meta_description": "car"
          },
          "slug": "car",
          "createdAt": "2023-07-05T08:51:02.701Z",
          "updatedAt": "2023-07-05T08:51:02.701Z"
        },
        {
          "id": "64a52eebf53cf502e2aa0cc7",
          "name": "Van",
          "archived": false,
          "meta": {
            "meta_title": "van",
            "meta_description": "van"
          },
          "slug": "van",
          "createdAt": "2023-07-05T08:50:51.644Z",
          "updatedAt": "2023-07-05T08:50:51.644Z"
        }
      ],
      "ref": "carvan",
      "location": "here",
      "typeOfDescription": "carvan",
      "manufacturer": {
        "id": "64a52f2af53cf502e2aa0d18",
        "name": "ME",
        "archived": false,
        "meta": {
          "meta_title": "me",
          "meta_description": "me"
        },
        "slug": "me",
        "createdAt": "2023-07-05T08:51:54.703Z",
        "updatedAt": "2023-07-05T08:51:54.703Z"
      },
      "registration": "asdf234",
      "operatingHours": 223,
      "mileageMiles": "3333",
      "fuel": "yes",
      "weight": "52",
      "horsePower": "1",
      "productCarousel": [
        {
          "image": {
            "id": "64a52f94f53cf502e2aa0d3f",
            "alt": "coeliac screenshot",
            "filename": "Photo on 09-05-2023 at 12.47.jpg",
            "mimeType": "image/jpeg",
            "filesize": 204410,
            "width": 1440,
            "height": 960,
            "sizes": {
              "card": {
                "width": 640,
                "height": 480,
                "mimeType": "image/jpeg",
                "filesize": 43876,
                "filename": "Photo on 09-05-2023 at 12.47-640x480.jpg",
                "url": "http://localhost:3000/media/Photo on 09-05-2023 at 12.47-640x480.jpg"
              },
              "normal": {
                "width": 1024,
                "height": 576,
                "mimeType": "image/jpeg",
                "filesize": 68340,
                "filename": "Photo on 09-05-2023 at 12.47-1024x576.jpg",
                "url": "http://localhost:3000/media/Photo on 09-05-2023 at 12.47-1024x576.jpg"
              },
              "feature": {
                "width": 1024,
                "height": 576,
                "mimeType": "image/jpeg",
                "filesize": 68340,
                "filename": "Photo on 09-05-2023 at 12.47-1024x576.jpg",
                "url": "http://localhost:3000/media/Photo on 09-05-2023 at 12.47-1024x576.jpg"
              }
            },
            "createdAt": "2023-07-05T08:53:40.737Z",
            "updatedAt": "2023-07-05T08:53:40.737Z",
            "url": "http://localhost:3000/media/Photo on 09-05-2023 at 12.47.jpg"
          },
          "id": "64a52f78ae1b9b4233f123c1"
        }
      ],
      "carousel": [],
      "carouselTitle": "nope",
      "carouselDescription": "nope",
      "richText": [
        {
          "children": [
            {
              "text": "nope"
            }
          ]
        }
      ],
      "slug": "carvan",
      "meta": {
        "meta_title": "asdfwww",
        "meta_description": "wefwdf"
      },
      "createdAt": "2023-07-05T08:53:53.896Z",
      "updatedAt": "2023-07-05T08:53:53.896Z"
    }
  ],
  "totalDocs": 1,
  "limit": 10,
  "totalPages": 1,
  "page": 1,
  "pagingCounter": 1,
  "hasPrevPage": false,
  "hasNextPage": false,
  "prevPage": null,
  "nextPage": null
}
```

## Graphql
``` graphql
query products {
  Products(where: { categories: { in: ["64a52ef6f53cf502e2aa0cd8"] } }) {
    totalDocs
  }
}

```
Result
``` json
{
  "data": {
    "Products": {
      "totalDocs": 0
    }
  }
}
```
